digicampipe/image/sky_image.py

```diff
@@ -0,0 +1,555 @@
+                        if j is not None:
+                            print('Selecting job id', j)
+                            solved_id = j
+                            image_id = i
+                            break
+                    time.sleep(5)
+                c.set_tag(image_id, tag)
+            else:
+                solved_id = existing_jobs['job_ids'][0]
+            while True:
+                stat = c.job_status(solved_id, justdict=True)
+                if stat is None:
+                    continue
+                print('Got job status:', stat)
+                if stat.get('status', '') in ['success', 'failure']:
+                    success = (stat['status'] == 'success')
+                    break
+                time.sleep(5)
+            retrieve_urls = []
+            if success:
+                wcs_file = os.path.join(tmpdir, 'stars.wcs')
+                corr_file = os.path.join(tmpdir, 'stars.corr')
+                retrieve_urls.append((apiurl.replace('/api/', '/wcs_file/%i' % solved_id), wcs_file))
+                retrieve_urls.append((apiurl.replace('/api/', '/corr_file/%i' % solved_id), corr_file))
+            for url, fn in retrieve_urls:
+                print('Retrieving file from', url, 'to', fn)
+                f = None
+                while f is None:
+                    try:
+                        f = urlopen(url)
+                    except HTTPError as err:
+                        print("ERROR", err, "getting url", url)
+                        f = None
+                        break
+                if f is None:
+                    continue
+                txt = f.read()
+                w = open(fn, 'wb')
+                w.write(txt)
+                w.close()
+                print('Wrote to', fn)
+            # coordinate system
+            self.wcs = None
+            self.reference_pixel = None
+            self.reference_ra_dec = None
+            self.cd_matrix = None
+            if success:
+                try:
+                    header_wcs = fits.open(wcs_file)[0].header
+                    self.wcs = WCS(header_wcs)
+                    # reference point
+                    self.reference_pixel = (header_wcs['CRPIX1'], header_wcs['CRPIX2'])
+                    self.reference_ra_dec = (header_wcs['CRVAL1'], header_wcs['CRVAL2'])
+                    # transformation matrix
+                    self.cd_matrix = np.array(((header_wcs['CD1_1'], header_wcs['CD1_2']),
+                                               (header_wcs['CD2_1'], header_wcs['CD2_2'])))
+                except FileNotFoundError:
+                    pass
+            # stars position in image
+            self.stars_pixel = None
+            if success:
+                try:
+                    stars_data = fits.open(corr_file)[1].data
+                    self.stars_pixel = np.array((stars_data['index_x'], stars_data['index_y']))
+                except FileNotFoundError:
+                    pass
```

Comment:
Same question as above. Why is it in this situation okay to let this error pass silently?

Reply:
That file contains stars expected in that region of the sky.
Only created if the coordinates were found.